resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/uri/URIUtils.java

```diff
@@ -57,6 +57,9 @@ public static String uriToString(NetworkInterfaceURI uri) {
     public static String getHost(String uriString) {
         try {
             URI uri = new URI(uriString);
+            if(uri.getHost()==null) {
+                throw new IllegalArgumentException("URI syntax invalid. Protocol, host and port are all required: " + uriString);
```

Comment:
The port is optional, so the message isn't quite correct. And I think the correct term is "scheme", not "protocol". How about something like this?:

> Invalid URI syntax. Scheme and host must be provided (port number is optional):